File: plot/src/main/java/net/fwitz/math/plot/binary/complex/ComplexFunctionPanel.java
package net.fwitz.math.plot.binary.complex;

import net.fwitz.math.binary.complex.Complex;
import net.fwitz.math.plot.binary.BinaryNumberFunctionPanel;

public class ComplexFunctionPanel extends BinaryNumberFunctionPanel<Complex, Complex> {
    ComplexFunctionPanel(ComplexFunctionPlot plot) {
        super(Complex.class, Complex.class, plot);
    }

    @Override
    protected ComplexFunctionRenderer<Complex> createRenderer(int width, int height) {
        return new ComplexFunctionRenderer<>(
                this, valueType,
                width, height,
                ComplexFunctionValuesFilters.getInstance());
    }
}

File: plot/src/main/kotlin/net/fwitz/math/plot/binary/complex/ComplexFunctionPanel.kt
package net.fwitz.math.plot.binary.complex

import net.fwitz.math.binary.complex.Complex
import net.fwitz.math.plot.binary.BinaryNumberFunctionPanel

class ComplexFunctionPanel internal constructor(
    plot: ComplexFunctionPlot
) : BinaryNumberFunctionPanel<Complex, Complex>(
    Complex::class.java,
    Complex::class.java,
    plot
) {
    override fun createRenderer(width: Int, height: Int): ComplexFunctionRenderer<Complex> {
        return ComplexFunctionRenderer(
            this, valueType,
            width, height,
            ComplexFunctionValuesFilters
        )
    }
}